collector/aws/elasticsearch.go

```diff
@@ -0,0 +1,290 @@
+	ListDomainNames(*elasticsearch.ListDomainNamesInput) (*elasticsearch.ListDomainNamesOutput, error)
+	ListTags(*elasticsearch.ListTagsInput) (*elasticsearch.ListTagsOutput, error)
+}
+
+// ElasticSearchManager will hold the ElasticSearch Manger strcut
+type ElasticSearchManager struct {
+	collector          collector.CollectorDescriber
+	client             ElasticSearchClientDescriptor
+	cloudWatchCLient   *CloudwatchManager
+	pricingClient      *PricingManager
+	stsClient          *STSManager
+	metrics            []config.MetricConfig
+	region             string
+	namespace          string
+	servicePricingCode string
+	Name               string
+}
+
+// DetectedElasticSearch defines the detected AWS Elasticsearch cluster
+type DetectedElasticSearch struct {
+	Metric        string
+	Region        string
+	InstanceType  string
+	InstanceCount int64
+	collector.PriceDetectedFields
+}
+
+// elasticSearchVolumeType will hold the available volume types for ESCluster EBS
+var elasticSearchVolumeType = map[string]string{
+	"gp2":      "GP2",
+	"standard": "Magnetic",
+	"io1":      "PIOPS",
+}
+
+// NewElasticSearchManager implements AWS GO SDK
+func NewElasticSearchManager(collector collector.CollectorDescriber, client ElasticSearchClientDescriptor, cloudWatchCLient *CloudwatchManager, pricing *PricingManager, sts *STSManager, metrics []config.MetricConfig, region string) *ElasticSearchManager {
+
+	return &ElasticSearchManager{
+		collector:          collector,
+		client:             client,
+		cloudWatchCLient:   cloudWatchCLient,
+		metrics:            metrics,
+		pricingClient:      pricing,
+		stsClient:          sts,
+		region:             region,
+		namespace:          "AWS/ES",
+		servicePricingCode: "AmazonES",
+		Name:               fmt.Sprintf("%s_elasticsearch", ResourcePrefix),
+	}
+}
+
+// Detect checks with elasticsearch cluster is underutilized
+func (esm *ElasticSearchManager) Detect() ([]DetectedElasticSearch, error) {
+
+	log.WithFields(log.Fields{
+		"region":   esm.region,
+		"resource": "elasticsearch",
+	}).Info("analyzing resource")
+
+	esm.collector.UpdateServiceStatus(collector.EventCollector{
+		ResourceName: esm.Name,
+		Data: collector.EventStatusData{
+			Status: collector.EventFetch,
+		},
+	})
+
+	detectedElasticSearchClusters := []DetectedElasticSearch{}
+
+	clusters, err := esm.DescribeClusters()
+	if err != nil {
+
+		esm.collector.UpdateServiceStatus(collector.EventCollector{
+			ResourceName: esm.Name,
+			Data: collector.EventStatusData{
+				Status:       collector.EventError,
+				ErrorMessage: err.Error(),
+			},
+		})
+
+		return detectedElasticSearchClusters, err
+	}
+
+	now := time.Now()
+
+	for _, cluster := range clusters {
+		log.WithField("cluster_arn", *cluster.ARN).Debug("checking elasticsearch cluster")
+
+		instancePrice, err := esm.pricingClient.GetPrice(esm.GetPricingFilterInput([]*pricing.Filter{
```

Comment:
It is very hard to read this line. 
i think you can move the pricing struct into a variable and it will be more clear what are you sending to `GetPrice`

Reply:
moved to a variable